src/test/java/seedu/address/logic/commands/CloseClaimCommandTest.java

```diff
@@ -0,0 +1,94 @@
+import seedu.address.model.client.insurance.claim.Claim;
+import seedu.address.model.client.insurance.claim.ClaimStub;
+
+class CloseClaimCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    void execute_validIndexUnfilteredList_success() throws CommandException,
+            InsurancePlanException, ClaimException {
+        ModelManager expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
+
+        Client clientToEdit = model.getFilteredClientList().get(INDEX_THIRD_CLIENT.getZeroBased());
+        InsurancePlan insurancePlan = clientToEdit.getInsurancePlansManager().getInsurancePlan(0);
+        Claim claim = clientToEdit.getInsurancePlansManager().getInsurancePlan(0).getClaim("A1001");
+
+        CloseClaimCommand closeClaimCommand =
+                new CloseClaimCommand(INDEX_THIRD_CLIENT, 0, claim.getClaimId());
+
+        // make sure all the messages only have client name inside the actual classes
+        // because when the message is created here before modification there can be issues with
+        // the string comparison of the client after deletion and before deletion.
+        String expectedMessage = String.format(CloseClaimCommand.MESSAGE_CLOSE_CLAIM_SUCCESS,
+                clientToEdit.getName().toString(), insurancePlan, claim.getClaimId());
+
+        // rebuilds a copy of the claims from string to prevent same-object manipulation
+        String insurancePlanString = insurancePlan.toString();
+        String claimsString = clientToEdit.getInsurancePlansManager().convertClaimsToJson();
+        InsurancePlansManager updatedInsurancePlanManager = new InsurancePlansManager(insurancePlanString);
+        updatedInsurancePlanManager.addAllClaimsFromJson(claimsString);
+        updatedInsurancePlanManager.closeClaim(insurancePlan, claim);
+
+        Client updatedClient = new Client(clientToEdit.getName(), clientToEdit.getPhone(), clientToEdit.getEmail(),
+                clientToEdit.getAddress(), updatedInsurancePlanManager, clientToEdit.getTags());
+
+        expectedModel.setClient(clientToEdit, updatedClient);
+
+        assertCommandSuccess(closeClaimCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    void execute_invalidClaim_exceptionHandled() {
+
+        Client clientToEdit = model.getFilteredClientList().get(INDEX_THIRD_CLIENT.getZeroBased());
+        try {
+            InsurancePlan insurancePlan = clientToEdit.getInsurancePlansManager().getInsurancePlan(0);
+        } catch (InsurancePlanException e) {
+            fail();
+        }
+
+        Claim fakeClaim = new ClaimStub();
+        CloseClaimCommand failingCloseClaimCommand =
+                new CloseClaimCommand(INDEX_THIRD_CLIENT, 0, fakeClaim.getClaimId());
+
+        assertThrows(CommandException.class, () -> failingCloseClaimCommand.execute(model));
+    }
+
+    @Test
+    void execute_invalidInsurancePlan_exceptionHandled() {
```

Comment:
Can consider renaming the method to be more intuitive (raised in the above comment)

Reply:
sure, no worries